src/Disks/ObjectStorages/IObjectStorage.h

```diff
@@ -66,7 +66,32 @@ class IObjectStorage
     virtual bool exists(const StoredObject & object) const = 0;
 
     /// List on prefix, return children (relative paths) with their sizes.
-    virtual void listPrefix(const std::string & path, RelativePathsWithSize & children) const = 0;
+    /// So it should return all "objects" with their sizes.
+    ///
+    /// For example if you do this over filesystem, you should skip folders and
+    /// return files only, so something like on local filesystem:
+    ///
+    ///     find . -type f
+    ///
+    /// NOTE: It makes sense only for real object storages (S3, Azure), since
+    /// it is used only for one of the following:
+    /// - send_metadata (to restore metadata)
+    ///   - see DiskObjectStorage::restoreMetadataIfNeeded()
+    /// - MetadataStorageFromPlainObjectStorage - only for s3_plain disk
+    virtual void listPrefix(const std::string & path, RelativePathsWithSize & children) const;
+
+    /// List on prefix, but it is limited only to current path, something like
+    /// on local filesystem:
+    ///
+    ///     find . -maxdepth 1 -type f
+    ///
+    /// @param path - path to list prefix in
+    /// @param children - list of child nodes
+    /// @param common_prefixes - list of common prefixes in this path, for unix
+    /// filesystem it is simply subfolders.
```

Comment:
To get list of subfolders you should use `-type d`, not `-type f`

Reply:
Actually it is even `find . -maxdepth 1`, folders goes to `common_prefixes` and files to `children`